@here/harp-datasource-protocol/lib/StringEncodedNumeral.ts

```diff
@@ -48,16 +48,38 @@ export const StringEncodedPixels: StringEncodedNumeralFormat = {
         return [Number(StringEncodedPixels.regExp.exec(encodedValue)![1])];
     }
 };
-export const StringEncodedHex: StringEncodedNumeralFormat = {
+const StringEncodedHex: StringEncodedNumeralFormat = {
     type: StringEncodedNumeralType.Hex,
     size: 3,
-    regExp: /#([0-9A-Fa-f]{1,2})([0-9A-Fa-f]{1,2})([0-9A-Fa-f]{1,2})/,
+    regExp: /\#([0-9A-Fa-f]{1,8})/,
     decoder: (encodedValue: string) => {
-        tmpColor.set(encodedValue).getHSL(tmpHSL);
+        const match = StringEncodedHex.regExp.exec(encodedValue)!;
+        const hex = match[1];
+        const size = hex.length;
+        // Note that we simply ignore alpha channel value.
+        // TODO: To be resolved with HARP-7517
+        if (size === 3 || size === 4) {
+            // #RGB or #RGBA
+            tmpColor.setRGB(
+                parseInt(hex.charAt(0) + hex.charAt(0), 16) / 255,
+                parseInt(hex.charAt(1) + hex.charAt(1), 16) / 255,
+                parseInt(hex.charAt(2) + hex.charAt(2), 16) / 255
+            );
+        } else if (size === 6 || size === 8) {
+            // #RRGGBB or #RRGGBBAA
+            tmpColor.setRGB(
+                parseInt(hex.charAt(0) + hex.charAt(1), 16) / 255,
+                parseInt(hex.charAt(2) + hex.charAt(3), 16) / 255,
+                parseInt(hex.charAt(4) + hex.charAt(5), 16) / 255
+            );
+        } else {
+            throw new Error(`unsupported hex color '${encodedValue}'`);
+        }
+        tmpColor.getHSL(tmpHSL);
```

Comment:
this test looks a bit too expensive, we do two regex tests. Please consider that we have 100s of interpolations that get evaluated for each frame and for each one the interpolations that is using literals as value we run the above tests.

Reply:
This is comming with a new patchset - single evaluation will be performed. I will update comment here when it will be available.

Reply:
Here is desired optimization, please review this PR to solve it (dependency).